packages/entity-server/src/routes/hierarchy/middleware/filter.ts

```diff
@@ -52,10 +51,45 @@ const toJsonBKey = <T extends keyof NestedFilterQueryFields>(nestedField: T): Js
   return [field, value].join(JSONB_FIELD_DELIMITER) as JsonBKey<T>;
 };
 
+// Inspired by Google Analytics filter: https://developers.google.com/analytics/devguides/reporting/core/v3/reference?hl=en#filters
+const operatorToSqlComparator = {
+  '==': null, // Exact match
+  '!=': '!=', // Does not match
+  '=@': 'ilike', // Contains sub string
+};
+const filterOperators = Object.keys(operatorToSqlComparator);
```

Comment:
```suggestion
type Operator = keyof typeof operatorToSqlComparator;

const filterOperators = Object.keys(operatorToSqlComparator) as Operator[];
```
`Object.keys()` is an area that typescript kinda drops the ball imo... It always returns type of `string[]` even tho it could theoretically return a string literal array like `('==' | '!=' | '=@')[]`. Oh well, best to cast it since we know what it'll be and then our types start playing nicely again 👍